src/components/Tank/Tank.tsx

```diff
@@ -0,0 +1,30 @@
+import React, { FC, ReactElement, useEffect, useRef } from 'react';
+import classNames from 'classnames';
+import { useGridObjectState } from '../Grid/Grid.helpers';
+import { getTankPositions, SPRITE_X_POS_VAR_NAME, SPRITE_Y_POS_VAR_NAME } from './Tank.helpers';
+import { TankState } from './Tank.model';
+import styles from './Tank.module.scss';
+
+const Tank: FC = (): ReactElement => {
+  const [state, inlineStyles] = useGridObjectState<TankState>();
+  const tankElRef = useRef<HTMLDivElement>(null);
+
+  const { xPos, yPos } = getTankPositions(state.color, state.position, state.type);
+
+  useEffect(() => {
+    const tankStyle: CSSStyleDeclaration | undefined = tankElRef.current?.style;
+    tankStyle?.setProperty(SPRITE_X_POS_VAR_NAME, xPos.toString());
+    tankStyle?.setProperty(SPRITE_Y_POS_VAR_NAME, yPos.toString());
+  }, [xPos, yPos]);
+
+  return (
+    <div
+      ref={tankElRef}
+      // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-unsafe-call
+      className={classNames([styles.Tank, state.onMovement && styles.onMovement])}
```

Comment:
- [ ] add `@types/classnames` ass dev dependency to not have problems with unsafe types.

Reply:
it worked :D